orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -677,21 +685,27 @@ class SqlExecutionRepository(
         )
         .fetchExecution()
 
-      if (execution != null) {
-        if (!execution.status.isComplete) {
-          return execution
-        }
-        jooq.transactional {
-          it.deleteFrom(table("correlation_ids")).where(field("id").eq(correlationId)).execute()
-        }
+      if (execution == null) {
```

Comment:
A POTENTIAL concern since we ARE querying now from the read pool:  Can the orchestration be on a delayed sync, so it'd actually exist in the RW & not be sync'd yet?  Wondering if a fallback is needed in such a case.  OR this could be handled upstream when it doesn't find it to retry?

Reply:
Yeah it's a concern - I know of some workflows that use orchestration stuff that could get impacted by this, and read-sync issues are the "biggest" headaches of RO queries based on past.  This seems to be the biggest spot that's a POTENTIAL risk as I THINK (and still tracing) the others do retries on an operation that throws an error or doesn't respond.  It's possible these upstream may ALSO retry, but not 100% sure YET if that exception is encountered...

Reply:
OK, reverted back to `withPool(poolName)`